### DISCUSSION TITLE
Are strings always null terminated?

### DISCUSSION THREAD

**Discussion by beleon** [janet-lang/janet#1129] [Q&A] [answered]
Janet stores the length of strings, so they don't need to be null terminated. I found that at least some strings are null terminated, but is this guaranteed or just a coincidence?

**Comment by beleon** [comment]
EDIT: I had a bug in my test setup, my original answer (below) is not correct, janet strings are in fact always null terminated.

No. After some testing I found out that janet strings are not guaranteed to be null terminated.

**Comment by ianthehenry** [comment, marked as answer]
`JANET_STRING`s are [always null-terminated](https://github.com/janet-lang/janet/blob/2ac4988f1bc4d292273a794e93ea8c386cb67523/src/core/string.c#L38), but they might contain null bytes inside their payload as well, so they aren't exactly C-string compatible. `JANET_BUFFER`s are not null-terminated.
